### ISSUE TITLE
varLib: allow no master at default location

### ISSUE THREAD

**Issue by moyogo** [fonttools/fonttools#1040]
On should be able to make a variable font with the default location at any instance, including one that has no master.

**Comment by twardoch** [comment]
This would be highly desirable. 

A typical scenario would be that I have a 2-axes, 4-master design space setup (say: wght & wdth where A is Light Wide, B is Black Wide, C is Light Narrow and D is Black Narrow): 

```
A----B
|    |
|    |
|    |
|    |
C----D
```

yet in the final font, I wish the neutral master to be a "Regular", so the final design space might look like: 

```
A-E--B
| |  |
| |  |
G-N--H
| |  |
C-F--D
```

If N is supposed to be the neutral master in the final font, the masters E F G H also need to be generated. 

Note how in a 2-axis scenario, when I want to declare the neutral master in an arbitrary position, the number of masters goes up from 4 to 9. In a 3-axis scenario it would go from 8 to 27. 

Also, since intermediate masters with min-max regions are not permitted to cut the zero axis, then if the neutral master is moved into an instance situated in a region affected by some intermediate masters, quite a few things need to be reconfigured as well (even more masters added). 

This is a noticeable deficiency of the OTVar model, but I agree with Denis that having an automated way of achieving this would be very desirable — especially since doing this in any manual manner would be indeed laborious.

**Comment by davelab6** [comment]
After generating EFGHN, can ABCD be removed? Since eg B is E+H

**Comment by behdad** [comment]
No. And that's not true.

**Comment by davelab6** [comment]
Ok cool, thanks for confirming. Fyi we have about 20 GF families set up AB or ABCD and can't generate them with fontmake

**Comment by moyogo** [comment]
@davelab6 could you report the errors or what is needed as issues on the fontmake repo?

**Comment by m4rc1e** [comment]
@moyogo sure!

**Comment by anthrotype** [comment]
The error in this cases should be this AssertionError
https://github.com/fonttools/fonttools/blob/ae73c45e6a8d27381020538a422964d24a3926af/Lib/fontTools/varLib/__init__.py#L813

**Comment by behdad** [comment]
FWIW to do this is equivalent to building your font then using the instancer to move the default.

**Comment by anthrotype** [comment]
> equivalent to building your font then using the instancer to move the default

indeed, but in order to build a VF to begin with, _some_ master has to be identified as the default, only then one can move it. So, there has to be a at least one master in the designspace that is "full", non "sparse".
